ocaml/xapi/xapi_host.ml

```diff
@@ -1346,9 +1346,23 @@ let apply_edition_internal  ~__context ~host ~edition ~additional =
     | Missing_connection_details -> raise Api_errors.(Server_error(missing_connection_details, []))
     | License_checkout_error s -> raise Api_errors.(Server_error(license_checkout_error, [s]))
   in
-  Db.Host.set_edition ~__context ~self:host ~value:edition';
-  let features = Features.of_assoc_list features' in
-  copy_license_to_db ~__context ~host ~features ~additional
+
+  let update_feature fname fenabled host =
+    Db.Feature.get_all_records ~__context |>
+      List.filter (fun (_, r) -> r.API.feature_name_label = fname && r.API.feature_host = host) |>
+    function
+    | [] -> Db.Feature.create ~__context ~uuid:(Uuid.to_string (Uuid.make_uuid ()))
```

Comment:
We create records here, and update them below, but where do we destroy them? Better would be to treat this as an operation on sets: take the set of features in the DB, and the set of updated features from v6d, and use set difference and intersection to find records that need to be created, updated, and destroyed.